src/app/clusters/access-control-server/access-control-cluster.cpp

```diff
@@ -639,19 +639,16 @@ std::optional<DataModel::ActionReturnStatus> AccessControlCluster::HandleReviewF
     CHIP_ERROR err = GetAccessControl().GetAccessRestrictionProvider()->RequestFabricRestrictionReview(
         commandObj->GetAccessingFabricIndex(), entries, token);
 
-    if (err == CHIP_NO_ERROR)
-    {
-        Clusters::AccessControl::Commands::ReviewFabricRestrictionsResponse::Type response;
-        response.token = token;
-        commandObj->AddResponse(commandPath, response);
-    }
-    else
+    if (err != CHIP_NO_ERROR)
     {
         ChipLogError(DataManagement, "AccessControlCluster: restriction review failed: %" CHIP_ERROR_FORMAT, err.Format());
         return Protocols::InteractionModel::ClusterStatusCode(err);
     }
 
-    return Protocols::InteractionModel::Status::Success;
+    Clusters::AccessControl::Commands::ReviewFabricRestrictionsResponse::Type response;
+    response.token = token;
+    commandObj->AddResponse(commandPath, response);
+    return std::nullopt;
```

Comment:
why is nullopt used instead of success? I suppose this isn't your code necessarily, but ... ?

Reply:
Is this a case where the responsibility for adding the status codes is split between the caller and the function?

Reply:
This is actually my code update: we _must_ return nullopt if we set the return ourselves. Previous code was returning success, which is wrong (it double-sets the status with conflicting/different values)